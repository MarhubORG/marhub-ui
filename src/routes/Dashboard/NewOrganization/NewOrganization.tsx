import React, { Component } from 'react';
import styled from 'styled-components';
import { navigate } from '@reach/router';
import { Dispatch } from 'redux';
import { connect } from 'react-redux';
import {
  CreateOrganizationAction,
  createOrganization,
  createOrganizationRedirect,
  CreateOrganizationRedirectAction,
} from '../../../redux/actions/dashboard';
import TextInput from '../../../components/Forms/TextInput/TextInput';
import { RootState } from '../../../types/interfaces';

interface NewOrganizationProps {
  message?: string;
  createOrganization(name: string): CreateOrganizationAction;
  createOrganizationRedirect(): CreateOrganizationRedirectAction;
  redirectToVisibleFields: string;
}

interface NewOrganizationState {
  name: string;
}

/* eslint-disable @typescript-eslint/indent */
export class UnconnectedNewOrganization extends Component<
  NewOrganizationProps,
  NewOrganizationState
> {
  /* eslint-enable @typescript-eslint/indent */

  constructor(props: NewOrganizationProps) {
    super(props);
    this.state = {
      name: '',
    };
  }

  handleChange = (name: string): void => {
    this.setState({ name });
  };

  handleClick = (): void => {
    this.props.createOrganization(this.state.name);
  };

  render(): JSX.Element {
    if (this.props.redirectToVisibleFields.length > 0) {
      this.props.createOrganizationRedirect();
      navigate(
        `/dashboard/organizations/organization-export-template/${this.props.redirectToVisibleFields}`
      );
    }
    return (
      <Layout>
        <div>{this.props.message}</div>
        <h1>New Organization</h1>
        <form>
          <TextInput
            htmlFor="name"
            labelText="Name:"
            placeholder="Name"
            name="name"
            value={this.state.name}
            onChange={this.handleChange}
          />
          <StyledButton
            type="button"
            disabled={this.state.name.length <= 0}
            onClick={this.handleClick}
          >
            Submit
          </StyledButton>
        </form>
      </Layout>
    );
  }
}

const StyledButton = styled.button`
  height: 2rem;
  width: 8rem;
  margin-top: 1rem;
  border-radius: 5px;
  background-color: ${({ theme }): string => theme.primaryColor};
  color: ${({ theme }): string => theme.white};
`;

const Layout = styled.div`
  padding: 1rem 2rem;
`;

export interface MapDispatchToProps {
  createOrganization(name: string): CreateOrganizationAction;
  createOrganizationRedirect(): CreateOrganizationRedirectAction;
}

export function mapDispatchToProps(dispatch: Dispatch): MapDispatchToProps {
  return {
    createOrganization: (name: string): CreateOrganizationAction =>
      dispatch(createOrganization(name)),
    createOrganizationRedirect: (): CreateOrganizationRedirectAction =>
      dispatch(createOrganizationRedirect()),
  };
}

export interface MapStateToProps {
  redirectToVisibleFields: string;
}

export function mapStateToProps(state: RootState): MapStateToProps {
  const { redirectToVisibleFields } = state.dashboardReducer;
  return { redirectToVisibleFields };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UnconnectedNewOrganization);
