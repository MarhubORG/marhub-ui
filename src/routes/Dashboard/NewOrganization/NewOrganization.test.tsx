import React from 'react';
import renderer from 'react-test-renderer';
import { UnconnectedNewOrganization } from './NewOrganization';
import {
  createOrganization,
  createOrganizationRedirect,
} from '../../../redux/actions/dashboard';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <UnconnectedNewOrganization
        createOrganization={createOrganization}
        createOrganizationRedirect={createOrganizationRedirect}
        redirectToVisibleFields=""
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
