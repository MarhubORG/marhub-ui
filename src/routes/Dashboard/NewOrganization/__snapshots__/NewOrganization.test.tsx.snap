// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c2 {
  height: 2rem;
  min-width: 18rem;
  max-width: 18rem;
  border-radius: 0.2rem;
  font-size: 0.9rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border: 1px solid #edf1f7;
}

.c1 {
  display: block;
  font-family: Open Sans,sans-serif;
  font-size: 0.75rem;
  font-weight: 700;
  line-height: 1rem;
  margin: 0.5rem 0rem;
}

.c3 {
  height: 2rem;
  width: 8rem;
  margin-top: 1rem;
  border-radius: 5px;
}

.c0 {
  padding: 1rem 2rem;
}

<div
  className="c0"
>
  <div />
  <h1>
    New Organization
  </h1>
  <form>
    <div>
      <label
        className="c1"
        htmlFor="name"
      >
        Name:
      </label>
      <input
        className="c2"
        name="name"
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value=""
      />
    </div>
    <button
      className="c3"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Submit
    </button>
  </form>
</div>
`;
